secrets/store.go

```diff
@@ -70,6 +80,18 @@ func (s *Store) parser(r io.Reader) (interface{}, error) {
 	return secrets, nil
 }
 
+func (s *Store) dirParser(r io.Reader) (interface{}, error) {
+	drc := r.(filewatcher.DummyReadCloser)
```

Comment:
```suggestion
	drc, ok := r.(filewatcher.DummyReadCloser)
	if !ok {
	  return nil, fmt.Errorf("secrets.Store.dirParser: invalid type: %T", r)
	}
```

I think we should check if the type assertion was successful. We might be able to guarantee the type but if the code ever changes it could change into a panic.

Reply:
good call, ty